pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -332,14 +332,15 @@ func TestKeepAnnotatedDeploymentAlwaysRemain(t *testing.T) {
 				AdvancedClusters: advancedClusterClient,
 			}
 			deployment := v1.NewDeployment(project.Namespace, fakeDeployment, fakeDeployment)
-			deployment.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
 			customresource.SetAnnotation(deployment,
 				customresource.ResourcePolicyAnnotation,
 				customresource.ResourcePolicyKeep,
 			)
 			k8sclient := testK8sClient()
 			customresource.SetFinalizer(deployment, customresource.FinalizerLabel)
 			require.NoError(t, k8sclient.Create(context.Background(), deployment))
+			// set deletion timestamp after creation in k8s
+			deployment.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
```

Comment:
This won't be applied to the resource in K8S, but only to the resource in the operator memory. Not sure if this is made intentionally.

Reply:
Yes and no. Yes I moved this here intentionally, as it was on line 335 before. The reason is that it seems that the creation step now will reset the deletion timestamp, as if it had never been set, sp the tests later would fail.

And no, it was not my intention for the timestamp to be only in memory, but for this test case it is fine. The important bit is that the reconciliation that runs later sees that deletion timestamp set and acts accordingly. So I do not think we need an update here.

Take away, seems the new library version will NOT allow you to create a k8s object that is already marked for deletion. It kind of makes sense, but the old library would respect the timestamp you set there.